src/tuple_gen.cpp

```diff
@@ -177,23 +181,62 @@ void CIpInfoBase:: detach_pool_and_client_node(void *ip_pool){
         }
     }
 }
-/* base on thread_id and client_index 
-  client index would be the same for all thread 
-*/
+
+uint64_t entropy64() {
+    std::random_device src;
+    return (uint64_t(src()) << 32) | src();
+}
+
+struct XorShift32Star final {
+    explicit XorShift32Star(uint64_t seed) : state(seed | 1) {} // state must be non-zero
+    explicit XorShift32Star() : XorShift32Star(entropy64()) {}
+
+    using result_type = uint32_t;
+
+    static constexpr result_type min() { return 0; }
+    static constexpr result_type max() { return result_type(~result_type(0)); }
```

Comment:
Somewhat simpler:

```suggestion
    static constexpr result_type min() { return +0; }
    static constexpr result_type max() { return -1; }
```

or more correct:

```suggestion
    static constexpr result_type min() { return std::numeric_limits<result_type>::min(); }
    static constexpr result_type max() { return std::numeric_limist<result_type>::max(); }
```

YMMV.

Reply:
Done